gui/default/syncthing/core/syncthingController.js

```diff
@@ -549,10 +564,28 @@ angular.module('syncthing.core')
                 $scope.connectionsTotal = data.total;
 
                 data = data.connections;
+                var currentAddresses = [];
                 for (id in data) {
                     if (!data.hasOwnProperty(id)) {
                         continue;
                     }
+                    if (Object.keys(data).length != $scope.devices.length) {
```

Comment:
This check is still done in every loop iteration, and might trigger in situations where the device in question would actually be available.

```suggestion
                    if (!(id in $scope.devices)) {
```